Sources/Fuzzilli/Corpus/MarkovCorpus.swift

```diff
@@ -0,0 +1,196 @@
+            print("Markov corpus requires the use of a ProgramCoverageEvaluator as its evaluator")
+            exit(-1)
+        }
+        self.genericProg = nil
+        super.init(name: "MarkovCorpus")
+
+    }
+
+    override func initialize() {
+        self.genericProg = makeSeedProgram()
+    }
+
+    public func add(_ program: Program, _ aspects: ProgramAspects) {
+        guard program.size > 0 else { return }
+        allIncludedPrograms.append(program)
+        if let covAspects = aspects as? CovEdgeSet {
+            let edges = covAspects.toEdges()
+            for e in edges {
+                edgeMap[e] = program
+            }
+        } else {
+            logger.fatal("Markov Corpus needs to be provided a CovEdgeSet when adding a program")
+        }
+    }
+
+    // Switch evenly between programs in the current queue and all programs available to the corpus
+    public func randomElementForSplicing() -> Program {
+        if(size <= 1) {
+            return genericProg!
+        }
+        var prog = programExecutionQueue.randomElement()
+        if prog == nil || probability(0.5) {
+            prog = allIncludedPrograms.randomElement()
+        }
+        assert(prog!.size != 0)
+        return prog!
+    }
+
+    /// For the first 500 executions, randomly choose a program. This is done to build a base list of edge counts
+    /// Once that base is acquired, provide samples that trigger an infrequently hit edge
+    public func randomElementForMutating() -> Program {
+        totalExecs += 1
+        if(size <= 1){
+            return genericProg!
+        }
+        // Only do computationally expensive work choosing the next program when there is a solid
+        // baseline of execution data. The data tracked in the statistics module is not used, as modules are intended 
+        // to not be required for the fuzzer to function.
+        if totalExecs > 1000 {
+            // Check if more programs are needed
+            if programExecutionQueue.count == 0 {
+                regenProgramList()
+            }
+            assert(programExecutionQueue.count > 0)
+            var prog : Program
+            if let tempProg = currentProg, remainingEnergy > 0 {
+                prog = tempProg
+                remainingEnergy -= 1
+            } else {
+                prog = programExecutionQueue.popLast()!
+                currentProg = prog
+                remainingEnergy = energyBase()
+            }
+            return prog
+        } else {
+            return allIncludedPrograms.randomElement()!
+        }
+    }
+
+    private func regenProgramList() {
+        assert(programExecutionQueue.count == 0)
+        let covEdgesBuffPtr = covEvaluator.getEdgeCountPtr()!
+        // Per https://developer.apple.com/documentation/swift/unsafebufferpointer, making an array from an UnsafeBufferPtr copies the memory
+        var covEdgesArr = Array(covEdgesBuffPtr)
```

Comment:
Nit: maybe call this something like `edgeCountsSorted` or so to avoid confusion with `covEdgesBuffPtr`?